### PR TITLE
Add Wasbs support to Wasb Underfs

### PR DESCRIPTION
**Pull request by imiller31**
Azure Blob Storage over https is currently unsupported by the `wasb` underfile system; however, the version of [azure-hadoop ](https://hadoop.apache.org/docs/r2.7.3/hadoop-azure/index.html) used in the underfs supports the https scheme (wasbs). 

This PR merely changes the accepted `SCHEME` from `wasb://` to accepted `SCHEMES` including both `wasb://` and `wasbs://`

`wasbs` support was tested by running `runUfsTests` in a k8s installation. All tests were passed successfully.

### CODE REVIEW

File: underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java [hunk 1]
```diff
@@ -36,15 +38,18 @@
...
   /**
    * Prepares the configuration for this Wasb as an HDFS configuration.
    *
    * @param conf the configuration for this UFS
+   * @param isWasb whether to use wasb configuration or not
    * @return the created configuration
    */
-  public static Configuration createConfiguration(UnderFileSystemConfiguration conf) {
+  public static Configuration createConfiguration(UnderFileSystemConfiguration conf,
+          Boolean isWasb) {
```
Comment (ZacBlanco) [inline underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: `isWasb` doesn't seem very clear to me. Do you think something like `isSecure` might be easier to understand? - If so, you'd also have to flip the logic for the conditional below

Comment (imiller31) [reply underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: Sure, I have changed this flag to `isSecure`.

File: underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java [hunk 2]
```diff
@@ -53,8 +58,13 @@ public static Configuration createConfiguration(UnderFileSystemConfiguration con
         wasbConf.set(key, value);
       }
     }
-    wasbConf.set("fs.AbstractFileSystem.wasb.impl", "org.apache.hadoop.fs.azure.Wasb");
-    wasbConf.set("fs.wasb.impl", "org.apache.hadoop.fs.azure.NativeAzureFileSystem");
+    if (isSecure) {
+      wasbConf.set("fs.AbstractFileSystem.wasb.impl", "org.apache.hadoop.fs.azure.Wasbs");
```
Comment (gpang) [inline underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: Does this not have to be "fs.AbstractFileSystem.wasbs.impl"?

File: underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java [hunk 3]
```diff
@@ -67,7 +77,7 @@ public static Configuration createConfiguration(UnderFileSystemConfiguration con
    */
   public static WasbUnderFileSystem createInstance(AlluxioURI uri,
       UnderFileSystemConfiguration conf) {
-    Configuration wasbConf = createConfiguration(conf);
+    Configuration wasbConf = createConfiguration(conf, uri.getScheme().startsWith("wasb://"));
```
Comment (gpang) [inline underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: Shouldn't this be `startsWith("wasbs://")`?

Instead, can we have 2 other constants, like `SCHEME_SECURE="wasbs://"`, and `SCHEME_INSECURE="wasb://"`?

File: underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java (hunk 2 above)
Comment (imiller31) [reply underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: Yes this should be, that is a good catch. Strange that `wasbs` worked correctly in my testing.

Comment (imiller31) [reply underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: I've validated the new code works on a k8s installation. Resolving.

File: underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java (hunk 3 above)
Comment (imiller31) [reply underfs/wasb/src/main/java/alluxio/underfs/wasb/WasbUnderFileSystem.java]: Changed and verified

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**